SalesforceCommerceCloud/pwa-kit
# Update linter rules for better extensibility

1. Sometimes VS Code automatically adds imports, but instead of doing importing from `foo`, it will import from `@salesforce/retail-react-app/node_modules/foo`, which is not something that we want. I added a linter rule to catch this. It's in the monorepo root because it only impacts the monorepo. (If anyone has a way to make VS Code _not_ do this, I'd also appreciate that as a solution!)

2. The rule to map replace relative imports with `@salesforce/retail-react-app` imports also only impacts monorepo development, so I moved that rule to the monorepo root as well. This also fixes `npm run lint` for generated projects that don't use extensibility. It was previously broken because the eslint plugin was specified as a dev dependency in `pwa-kit-dev`, and therefore wasn't installed in generated projects.

# Description



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- Paste this in any JS file:
```js
import {useIntl} from '@salesforce/retail-react-app/node_modules/react-intl/index'
```
- Linter should report an error!

- Generate a new project with `create-mobify-app-dev.js` that does _not_ use extensibility.
- Executing `npm run lint` in the generated file should work.

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
